feat(zombies): add zombie-virus and improve consume

kemperhun opened the pull request:
Добавлен Concealment Syndrome - Не позволяет узнать или намутировать радиацией антиген вируса.
Добавлен HQ41 - превращает пациента в зомби
В качестве мажорной эпидемии добавлен зомби-вирус. Симптомы подобраны согласно #9351 
Пожирание зомбей теперь восстанавливает им сытость и заражает жертву всем букетом зомби-болячек.

<details>
<summary>Чейнджлог</summary>

```yml
🆑
tweak: Добавлена эпидемия зомби-вируса.
bugfix: Теперь зомби насыщаются, когда едят других людей.
/🆑
```

</details>

- [x] Pull Request полностью завершен, мне не нужна помощь чтобы его закончить.
- [x] Я внимательно прочитал все свои изменения и багов в них не нашел.
- [x] Я запускал сервер со своими изменениями локально и все протестировал.
- [x] Я ознакомился c [Guide to Contribute](https://github.com/ChaoticOnyx/OnyxBay/blob/dev/docs/contributing.md).

kemperhun committed:
Squashed commit of the following:

commit 0feb3a132f3b207765bc2062df9a4d20604258b8
Author: TobyThorne <45202681+TobyThorne@users.noreply.github.com>
Date:   Mon May 23 19:44:01 2022 +0600

    fix(antagonist): fixes antag icons

    PR #8443

commit 70e26b90385cc86c2e55aa702c9fc78c6880b26c
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Mon May 23 00:19:05 2022 +0300

    tweak(polar): extend map and cargo rework

    #8438

commit 3dce266d57c11c404cb7d86e6b4e8129cb7412ab
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Sun May 22 19:11:27 2022 +0300

    update(changelog): 22.5.2022

    #8446

commit 2e9deca40c9aa2d0b9e1487e56fa64e982cb7d57
Author: MaxZ043 <98965002+MaxZ043@users.noreply.github.com>
Date:   Sun May 22 16:28:40 2022 +0300

    refact(lipstick): non-hardcoded lipstick colors, dynamic layering

    #8367

commit 251f1898f0c951f037a67451ee989bec811af2c0
Author: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>
Date:   Sun May 22 14:39:23 2022 +0300

    fix(crew_monitor): proper use of map traits

    PR #8440

commit df8f30cbed8502debfcf7b50dc9827d4701a367f
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Sun May 22 14:36:33 2022 +0300

    fix(maps): fix species restricts

    PR #8430

commit f9cb88959fedb9184da9cba05fe1f07996e15857
Author: No-Tips <75416333+No-Tips@users.noreply.github.com>
Date:   Sat May 21 19:48:10 2022 +0300

    map(polar): remove pipe n wire madness

    #8420

commit 3498e5023f9d864f306fab1e4a9092dfbeaa36fe
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Sat May 21 19:47:52 2022 +0300

    fix(console): add missing camera networks

    #8426

commit 1d107402236bf53cf277a8b1dd79418315f879b2
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Sat May 21 19:47:32 2022 +0300

    fix(genesis): add void suits and rig for nukers

    #8427

commit 995ec36ec4124d074dd0db1aae1193dd04baddc6
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Sat May 21 17:30:59 2022 +0300

    fix(genesis): add missing bomb

    #8422

commit dc25b6df112fef2f45ec149f45c12fd11d058faf
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Sat May 21 18:09:37 2022 +0600

    update(changelog): 21.5.2022 (#8421)

    Co-authored-by: github-actions <github-actions@github.com>

commit 85899edce48f60bbd37b05d86a06b74984e71340
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Sat May 21 15:06:35 2022 +0300

    fix(console): fixes security camera map

    PR #8407

commit dea6418bc40a0055da45e994092a94c82603d99a
Author: TermService <TermService@users.noreply.github.com>
Date:   Sat May 21 14:56:50 2022 +0300

    fix(signaler): signaler works inside assemble

    #8411

commit 50c115a42d7bd0bbd5405a2ab38e8ba3e3209f9c
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Sat May 21 14:20:39 2022 +0300

    fix(genesis): fix missing landmarks

    #8418

commit 7b8289425e44ac073120825d4191d77af1bca336
Author: Doster <63081538+Doster-d@users.noreply.github.com>
Date:   Sat May 21 13:47:00 2022 +0300

    tweak(admin): fix runtime on server startup

    PR #8413

commit 647ca1a0dc576b63b032df34dfc08dbe95ed1a12
Author: Doster <63081538+Doster-d@users.noreply.github.com>
Date:   Sat May 21 13:46:36 2022 +0300

    tweak(genesis): now block xenos from jobs

    PR #8417

kemperhun committed:
feat(zombies): add zombie-virus and improve consume()

Добавлены симптомы HQ41 и Concealment Syndrome
В ивенты добавлен зомби-вирус
Пожирание теперь утоляет голод зомби и инфицирует жертву

ArmenianOnyx commented:
Так, мне кажется ты Ддостера без денег оставил?

kemperhun committed:
fix consume proc

kemperhun committed:
another fix

No-Tips commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -470,3 +470,40 @@
 			skin_state = SKIN_NORMAL
 	update_skin(1)
 
+/mob/living/carbon/human/consume()
```
Вот тут накосячил, брат. Это че, верб?

No-Tips replied:
А,прок, ну, короче, ты его проком не обозвал
```suggestion
/mob/living/carbon/human/proc/consume()
```

No-Tips commented on code/modules/mob/living/carbon/human/human_powers.dm:
```diff
@@ -470,3 +470,40 @@
+	set category = "Abilities"
+
+	if (last_special > world.time)
+		to_chat(src, "<span class='warning'>You aren't ready to do that! Wait [round(last_special - world.time) / 10] seconds.</span>")
```
Спаны поправь.

kemperhun committed:
Пофиг

Все равно консьюм никем, кроме зомби, в билде не используется. Зато ошибку выдавать перестанет(наверное...)

kemperhun committed:
Спаны

спаны

kreeperHLC commented:
> Так, мне кажется ты Ддостера без денег оставил?

Пусть сначала геймдизайн пройдёт

kemperhun committed:
Squashed commit of the following:

commit c64fe5faa93f25cc5b2bd330118b8212330b99b4
Author: TobyThorne <45202681+TobyThorne@users.noreply.github.com>
Date:   Fri Sep 30 16:21:00 2022 +0600

    fix(gripper): fixes interactions

    PR #9547

commit f83b957f540cd5cd9b638b95c7ec50ceefa476d3
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Fri Sep 30 15:00:36 2022 +0600

    update(changelog): 30.9.2022 (#9546)

    Co-authored-by: github-actions <github-actions@github.com>

commit ddd342d853083cbee775bb915a237b8d54ba41ab
Author: TobyThorne <45202681+TobyThorne@users.noreply.github.com>
Date:   Fri Sep 30 14:55:16 2022 +0600

    fix(inventory): fixes cringey drop-related leftovers

    PR #9544

commit 9fe0952b6bcf672f50230d796d633cb4400f3c99
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Fri Sep 30 11:50:15 2022 +0300

    fix(abductors): removes improper from the name

    PR #9539

commit ca5145fe6a2282e1d2cf281af205c7b390e8fa03
Author: No-Tips <75416333+No-Tips@users.noreply.github.com>
Date:   Fri Sep 30 11:39:22 2022 +0300

    fix(barrel): sets density

    PR #9530

commit 4465b3d8c670e98f724413fc8626edaf025a746e
Author: Medji <75128764+Medjic@users.noreply.github.com>
Date:   Fri Sep 30 11:28:02 2022 +0300

    fix(icons): adds held icons for laptops and freezers

    PR #9543

commit a08a99e0357b152c4822f995b67209541edd5f88
Author: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Date:   Thu Sep 29 19:16:09 2022 +0300

    update(changelog): 29.9.2022

    PR #9523
    Co-authored-by: github-actions <github-actions@github.com>

commit 2b1f3bddfc5aebad7c57c43f865119547515d3c2
Author: MaxZ043 <98965002+MaxZ043@users.noreply.github.com>
Date:   Thu Sep 29 19:14:36 2022 +0300

    fix(windows): fixes the Great Silicate Runtime

    PR #9502

commit d231190846c09f63e4be934bc0bf9ea843545304
Author: TatarTheBuilder <51989098+TatarTheBuilder@users.noreply.github.com>
Date:   Thu Sep 29 19:14:03 2022 +0300

    tweak(Polar): changes stuff

    PR #9520

commit 00e90debfa9c078f89bb73c6526db156a6f21493
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Thu Sep 29 19:11:03 2022 +0300

    refact(runechat): remove quality (no) timers copypasta

    #9522

commit faa486324e9ca7a609c7dfaab6ee265bc4739eb0
Author: TobyThorne <45202681+TobyThorne@users.noreply.github.com>
Date:   Thu Sep 29 22:10:13 2022 +0600

    tweak(inventory): purges dumb procs, enforces drop()

    PR #9518

commit 0c1b67d346cb328749d82b12ffea92e56a944d43
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Thu Sep 29 18:54:53 2022 +0300

    fix(runtime): fix runtime with zero thinking time

    #9521

commit a3ea3f32463c81187a1d7999e75f680d5141f96b
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Wed Sep 28 18:11:26 2022 +0300

    feat(events): add flexible storyteller-friendly events system

    #9396

commit ae7b1bc59cb7f9369e9cc02361b7c21256102d1c
Author: TobyThorne <45202681+TobyThorne@users.noreply.github.com>
Date:   Wed Sep 28 21:09:54 2022 +0600

    tweak(machinery): resprites kitchen equipment

    PR #9512

commit a8d18a5cb74bb6e2f33cd8efc8b0740c7f2e73e4
Author: Doster <63081538+Doster-d@users.noreply.github.com>
Date:   Wed Sep 28 17:18:10 2022 +0300

    tweak(mod_list): add option to remove nulls

    PR #9312
    - Добавлена возможность удалить nullы из списка.

commit 742783538f3b74d874fb7f44ca04e4c7e6ec5db7
Author: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>
Date:   Wed Sep 28 13:03:29 2022 +0300

    fix(floors): fixes monotile floor construction

    PR #9498

commit 963a1938d3bc9bf2a1ae0ac9193cb6105628383a
Author: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>
Date:   Wed Sep 28 13:02:44 2022 +0300

    fix(map): fixes polar level-1 mapgen

    PR #9500

commit 86de1fc418e785a4d4d347fc748bb918b830b824
Author: No-Tips <75416333+No-Tips@users.noreply.github.com>
Date:   Wed Sep 28 12:45:22 2022 +0300

    fix(coctails): fixes coctails glass_required

    PR #9501

commit 737161f51312caef2d7eb123353d77c103d82fc2
Author: MaxZ043 <98965002+MaxZ043@users.noreply.github.com>
Date:   Tue Sep 27 23:07:03 2022 +0300

    fix(window_frame): added ability to reinforce & repair glasspanes with silicate

    PR #9433
    close #7555
    - Продвинутые рамы теперь можно чинить и укреплять силикатом.

commit 317a04c423a1650b4719f816bd1acda3bb1fdf5a
Author: Medji <75128764+Medjic@users.noreply.github.com>
Date:   Tue Sep 27 22:54:26 2022 +0300

    tweak(punk's jackets): fix jacket's for space punk's

    PR #9468
    - Поправил куртки для всех бодитипов и добавил возможность расстёгивать каждую куртку.

commit f162a6d9977e2aa247164f6d958d537f79173efb
Author: TatarTheBuilder <51989098+TatarTheBuilder@users.noreply.github.com>
Date:   Tue Sep 27 20:27:04 2022 +0300

    tweak(map): polar-4 level overhaul

    PR #9499

commit 8010affeb80b2550bf69390dc535a260f9a7bd4f
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Mon Sep 26 20:23:25 2022 +0300

    fix(runechat): add client's existential crisis check

    #9497

commit 2c1e688e88887c5e8b2ffd89649c5d24c796e11b
Author: Tsurupeta <41485301+Tsurupeta@users.noreply.github.com>
Date:   Mon Sep 26 16:32:32 2022 +0300

    fix(map): replaces exoplanet floor with asteroid

    PR #9496

commit f9a0ae5b2e21b1975948f57e669305aaa927a7d4
Author: oleggorru <oleggorru@gmail.ru>
Date:   Mon Sep 26 16:05:10 2022 +0300

    fix(xenoarcheaology): fix suspention generator

    PR #9352
    fix #8698
    - Археологический генератор щита больше не прикручивается к космосу.

commit 85443c64d406ec6432a0477e7efaa4742b69b1f4
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Mon Sep 26 16:02:31 2022 +0300

    feat(sounds): add footsteps to every turf

    PR #9469
    - Добавлены звуки шагов ко многим видам пола.

commit be1fbfff5b80abb1380151ee9c6ba915abc95df4
Author: Grigoriy <83020228+EnderGolem@users.noreply.github.com>
Date:   Mon Sep 26 15:46:45 2022 +0300

    fix(wizard): fixed teleport recharge, teleport familiar, goat ability

    PR #9239
    close #8790
    close #9036
    close #8263
    - После отмены телепорта скилл не сбрасывается в кд;
    - Исправлено, что козел не мог использовать свою способность;
    - Исправлено, что фамильяр телепортирует мастера  к себе;
    - Теперь мастер сам решает, когда телепортировать фамильяра.

commit 777e2082491b2d8de5dba3f71519cf96d887cd78
Author: Oubi256 <98897017+Oubi256@users.noreply.github.com>
Date:   Mon Sep 26 15:30:40 2022 +0300

    feat(wizard): reworked Blood Shard skill

    PR #9054
    fix #9003
    - Отныне умение применяется на самом маге, тратя его кровь в зависимости от уровня прокачки силы умения. Имеет аудио и анимацию применения. Сразу же после каста станит мага на секунду.

commit bdf5bed71021f288d998be7ae0e8f8e48526b96d
Author: Doster <63081538+Doster-d@users.noreply.github.com>
Date:   Mon Sep 26 15:28:05 2022 +0300

    fix(radio): Runtime in game.dm,216: Cannot execute null.can hear radio()

    PR #9311

commit 67853171b39564e6f7fb74b00b96d357eb1c4ea8
Author: MaxZ043 <98965002+MaxZ043@users.noreply.github.com>
Date:   Mon Sep 26 15:27:08 2022 +0300

    fix(fbp-icons): added eye icons & missing body builds

    PR #9464
    close #9467
    - Пока в иссуе #9462 происходит полная паронормальщина - продолжаем трогать глаза и незаметно исправлять баги;
    - При разработке прошлого проекта не учел force_icon, который насильно заменяет путь до иконки, вследствие чего выпускаю патч, добавляющий всем ФБП правильно расположенные глаза;
    - В проект так же включил обновленные спрайты ФБП под новые бодибилды, отрисовал пару ФБП под толстых персонажей.

commit 640ef6e38f09823103d0b9f60d3965d2bc88930a
Author: MaxZ043 <98965002+MaxZ043@users.noreply.github.com>
Date:   Mon Sep 26 15:23:57 2022 +0300

    tweak(Clown PDA): Clown PDAs remain static while being slipped on

    PR #9480
    - ПДА клоуна больше не отлетает аки мыло.

commit a1c407b4a2ff734387c4a5bae92f6f544c87ef02
Author: Medji <75128764+Medjic@users.noreply.github.com>
Date:   Mon Sep 26 15:22:20 2022 +0300

    tweak(leather coat's): New coat and fix old

    PR #9473

commit 86281181e6ea51b1f76f540eda03bee38072e2ba
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Mon Sep 26 15:15:49 2022 +0300

    log(runechat): log more detalis in invalid mob cases

    PR #9491

commit e3375958be503856bdd0bb6dc458e2c3513ff3c0
Author: MaxZ043 <98965002+MaxZ043@users.noreply.github.com>
Date:   Mon Sep 26 15:15:17 2022 +0300

    tweak(circuit): added circuit description

    PR #9471
    - QOL фича, добавляющая в описание к плате компоненты, которые требуются для сборки соответствующей машины.

commit 65cb5e51b7469d45638d61279afb93cd1b8b7560
Author: goliath <kreeper3211225@gmail.com>
Date:   Mon Sep 26 15:08:55 2022 +0300

    Add(asteroid): add two new z-levels to asteroid

    PR #9463
    #185
    - Добавляет ещё два уровня астероида, пока что отличаются только зоной;
    - Добавляет лестницы и склоны;
    - Оверлеи пещерных стен теперь воспринимают пустоту как пустоту для отрисовки.

commit 42a21469f848c0b5eb4b888838ca473f9648a788
Author: No-Tips <75416333+No-Tips@users.noreply.github.com>
Date:   Mon Sep 26 14:57:37 2022 +0300

    feat(config): add mapping section

    #9492

commit f31060aa171419e93fc4293148e5b29ea3375c94
Author: Игорь Спичкин <55196698+igorsaux@users.noreply.github.com>
Date:   Mon Sep 26 14:51:10 2022 +0300

    tweak(polar): polar tweaks

    #9470

commit b2f736aca4d6661da90b87a8faefe944ff6e2b3c
Author: No-Tips <75416333+No-Tips@users.noreply.github.com>
Date:   Sun Sep 25 21:37:15 2022 +0300

    map(frontier): add biodome varitations

    #9482

kemperhun committed:
Merge branch 'dev' into Zombievirus

TobyThorne commented:
епик пулл фейл
